### DISCUSSION TITLE
user-settable Front / Rear ports

### DISCUSSION THREAD

**Discussion by vbarelds** [netbox-community/netbox#7790] [Ideas]
At this moment the port types for front and rear ports are not user-settable. (**Q/A #7782**)

Of course you can choose any port type, but then you can rely on the port type as a "source of truth" .

So it would be handy to be able to add port types

**Comment by candlerb** [comment]
I'm afraid that's been rejected multiple times.  See #4371 and the various issues linked from that.

**Reply by DanSheps** [reply]
~~This might be more tenable now that we are not using id's and instead using strings.  Could possibly allow plugins to define their own with the caveat that they need to include a custom prefix in the "slug" and maybe something to indicate it is a custom interface/port in the name itself.

It is worth a FR if someone can come up with a proper implementation~~

After some review, we probably would not take this up either way.

**Reply by jeremystretch** [reply]
Unless you are manufacturing bespoke hardware, there should never be a reason to do this. Please see the [FAQ](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#how-can-i-add-new-interface-types).
